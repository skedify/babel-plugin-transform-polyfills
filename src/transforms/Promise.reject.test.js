import compile from '../../test/utils/compile'

describe('Promise.reject', () => {
  it('should replace Promise.reject with a polyfill', () => {
    const result = compile(
      `
      const c = Promise.reject()
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      const e = Promise.reject(Promise.reject([1, 2]))
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when part of a call chain', () => {
    const result = compile(
      `
      const c = Object.keys(Promise.reject())
    `
    )

    expect(result).toMatchSnapshot()
  })
})
