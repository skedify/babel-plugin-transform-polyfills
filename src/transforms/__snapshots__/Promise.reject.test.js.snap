// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promise.reject should not import polyfill more than once 1`] = `
"import _Promise_reject from \\"babel-plugin-transform-polyfills/polyfills/Promise/reject\\";

const e = _Promise_reject(_Promise_reject([1, 2]));"
`;

exports[`Promise.reject should replace Promise.reject with a polyfill 1`] = `
"import _Promise_reject from \\"babel-plugin-transform-polyfills/polyfills/Promise/reject\\";

const c = _Promise_reject();"
`;

exports[`Promise.reject should work when part of a call chain 1`] = `
"import _Promise_reject from \\"babel-plugin-transform-polyfills/polyfills/Promise/reject\\";
const c = Object.keys(_Promise_reject());"
`;
